app/Http/Controllers/EntryController.php

```diff
@@ -2,26 +2,38 @@
 
 namespace App\Http\Controllers;
 
+use App\Repositories\DonationRepository;
+use App\Repositories\NeedsRepository;
 use Illuminate\Http\Request;
 
 class EntryController extends Controller
 {
-    // private $need;
+    private $donation;
+    private $need;
 
     /*
      * EntryController constructor.
      */
-    public function __construct()
+    public function __construct(DonationRepository $donationRepository, NeedsRepository $needsRepository)
     {
+         $this->donation = $donationRepository;
+         $this->need = $needsRepository;
     }
 
     /**
      * List all donatiaons
      *
      * @return $this
      */
-    public function view()
+    public function view($type, $id)
```

Comment:
I don't know PHP that much, Laravel even less. Just out of curiosity, doesn't this need a corresponding method call change that's missing from this PR? Or is that handled through an implicit call somewhere?